### PR TITLE
Implement support for CAP guestsOrExternalUsers

### LINKED ISSUE #261: Add support for conditional access policy guestsOrExternalUsers settings
Support `conditions.users.includeGuestsOrExternalUsers` and `conditions.users.excludeGuestsOrExternalUsers` fields on conditional access policy.

See
* https://learn.microsoft.com/en-us/graph/api/resources/conditionalaccessusers?view=graph-rest-1.0
* https://learn.microsoft.com/en-us/graph/api/resources/conditionalaccessguestsorexternalusers?view=graph-rest-1.0

### PR DESCRIPTION
**Pull request by agileknight**
Fixes #261

### CODE REVIEW

**Comment by agileknight** [comment]
Work in progress PR where the changes are intended to be used: https://github.com/hashicorp/terraform-provider-azuread/pull/1222

File: msgraph/models.go [hunk 1]
```diff
@@ -693,12 +693,66 @@ type ConditionalAccessSessionControls struct {
...
+	IncludeGroups                *[]string                               `json:"includeGroups,omitempty"`
+	ExcludeGroups                *[]string                               `json:"excludeGroups,omitempty"`
+	IncludeRoles                 *[]string                               `json:"includeRoles,omitempty"`
+	ExcludeRoles                 *[]string                               `json:"excludeRoles,omitempty"`
+	IncludeGuestsOrExternalUsers *ConditionalAccessGuestsOrExternalUsers `json:"includeGuestsOrExternalUsers"`
+	ExcludeGuestsOrExternalUsers *ConditionalAccessGuestsOrExternalUsers `json:"excludeGuestsOrExternalUsers"`
+}
+
+type ConditionalAccessGuestsOrExternalUsers struct {
+	GuestOrExternalUserTypes *[]ConditionalAccessGuestOrExternalUserType `json:"guestOrExternalUserTypes,omitempty"`
```
Comment (manicminer) [inline msgraph/models.go:707]: When we're using a custom marshaler and the field type is different, we should tag it to be ignored by default, then it won't accidentally get marshaled to the wrong type if someone embeds it.

```suggestion
	GuestOrExternalUserTypes *[]ConditionalAccessGuestOrExternalUserType `json:"-"` // see ConditionalAccessGuestsOrExternalUsers.MarshalJSON / ConditionalAccessGuestsOrExternalUsers.UnmarshalJSON
```

File: msgraph/models.go [hunk 2]
```diff
@@ -693,12 +693,66 @@ type ConditionalAccessSessionControls struct {
...
+	ExternalTenants          *ConditionalAccessExternalTenants           `json:"externalTenants,omitempty"`
+}
+
+type ConditionalAccessExternalTenants struct {
+	MembershipKind *string   `json:"membershipKind,omitempty"`
+	Members        *[]string `json:"members,omitempty"`
```
Comment (manicminer) [inline msgraph/models.go:713]: Missed the type alias

```suggestion
	MembershipKind *ConditionalAccessExternalTenantsMembershipKind `json:"membershipKind,omitempty"`
	Members        *[]string                                       `json:"members,omitempty"`

```

**Review by manicminer — APPROVED** [review]
Many thanks for this @agileknight! Just a couple suggestions below, which I'll go ahead and add, this otherwise LGTM 🦖

**Comment by agileknight** [comment]
@manicminer I created a follow-up PR after more thorough testing: https://github.com/manicminer/hamilton/pull/264

**Pull request merged**